application/controllers/QuestionGroupsAdministrationController.php

```diff
@@ -100,47 +103,31 @@ public function actionView(int $surveyid, int $gid, $landOnSideMenuTab = 'struct
         }
 
         if (!Permission::model()->hasSurveyPermission($surveyid, 'surveycontent', 'read')) {
-                App()->user->setFlash('error', gT("Access denied"));
-                $this->redirect(App()->request->urlReferrer);
+            App()->user->setFlash('error', gT("Access denied"));
+            $this->redirect(App()->request->urlReferrer);
         }
-
+        $aData = $this->setSurveyIdAndObject([], $surveyid);
+        $aData['gid'] = $gid;
+        $aData['condarray'] = getGroupDepsForConditions($surveyid, "all", $gid, "by-targgid");
         //check if group with the gid exists
-        $questionGroup = QuestionGroup::model()->findByPk($gid);
-        if ($questionGroup === null) { //group does not exists ...
+        try {
+            $questionGroupService = $this->getQuestionGroupServiceClass();
+            $aData['oQuestionGroup'] = $oQuestionGroup = $questionGroupService->getQuestionGroupObject($surveyid, $gid);
+        } catch (Exception $e) {
```

Comment:
If you catch an exception : youn can not be sure the error was "Question group does not exists"

Must use $e , no ?

Reply:
The exception error message could be not so user friendly. Of course we can not be 100% sure what exception was thrown. But the most obvious one is the NotFoundException

Reply:
I don't look fiurther, but why not use something similar than model->getErrors ?